core/controllers/admin.py

```diff
@@ -70,6 +71,22 @@ def get(self):
         self.render_template('admin-page.mainpage.html')
 
 
+class AdminSessionHandler(webapp2.RequestHandler):
+    """Handler for creating and registering the user"""
+
+    def get(self):
+        auth_claims = auth_services.get_auth_claims_from_request(self.request)
+        if auth_claims is None:
+            auth_services.establish_auth_session(self.request, self.response)
+            self.redirect(self.request.uri)
+        else:
+            user_id = user_services.create_new_user(
```

Comment:
Check to see that the user doesn't exist first, that way we can make this idempotent. For example:
```python
user_settings = user_services.get_user_settings_by_auth_id(auth_claims.auth_id)
if user_settings is not None:
    return